### PR TITLE
Chore: Remove several 'as' type assertions

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Removes a handful of instances where 'as' type assertions are used.

**Which issue(s) this PR fixes**:
See #44430

### CODE REVIEW

File: public/app/features/comments/CommentView.tsx [hunk 1]
```diff
@@ -28,8 +28,9 @@ export const CommentView = ({ comments, packetCounter, addComment }: Props) => {
   }, [packetCounter]);
 
   const onUpdateComment = (event: FormEvent<HTMLTextAreaElement>) => {
-    const element = event.target as HTMLInputElement;
-    setComment(element.value);
+    if (event.target instanceof HTMLInputElement) {
+      setComment(event.target.value);
+    }
```
Comment (hugohaggmark) [inline public/app/features/comments/CommentView.tsx:33]: I think this change might actually change the implementation as it is today, right?

**Review by hugohaggmark — APPROVED** [review]
Lovely 🎉

File: public/app/features/comments/CommentView.tsx (hunk 1 above)
Comment (kaydelaney) [reply public/app/features/comments/CommentView.tsx:33]: You're right. Maybe something for a separate PR.

**Review by ifrost — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
quick, someone tell @joshhunt 😍

**Review by nmarrs — APPROVED** [review]
LGTM!

**Pull request merged**